### PR TITLE
Add support for `GDAL_HTTP_VERSION=2PRIOR_KNOWLEDGE`

### LINKED ISSUE #10361: `CURL_HTTP_VERSION_2_PRIOR_KNOWLEDGE` support
### Feature description

`GDAL_HTTP_VERSION=2` will set `CURLOPT_HTTP_VERSION` to `CURL_HTTP_VERSION_2_0`, but there's not support for `CURL_HTTP_VERSION_2_PRIOR_KNOWLEDGE` (and `CURL_HTTP_VERSION_3`, but that's a different story, I suppose). This can be useful with misconfigured servers that don't support ALPN and don't send the `Alt-Svc` headers.

### Additional context

https://curl.se/libcurl/c/CURLOPT_HTTP_VERSION.html

The docs imply that it makes no difference for non-HTTPS URLs, but that doesn't seem to be the case in practice, at least for `curl --http2-prior-knowledge`.

### PR DESCRIPTION
**Pull request by lnicola**
## What does this PR do?
Adds `GDAL_HTTP_VERSION=2PRIOR_KNOWLEDGE`, which may allow forcing HTTP/2 on servers which don't advertise support for it.

## What are related issues/pull requests?
Closes #10361

## Tasklist

 - [ ] Check whether it actually works
 - [ ] Make sure code is correctly formatted (cf [pre-commit configuration](https://gdal.org/development/dev_practices.html#commit-hooks))
 - [ ] Add test case(s)
 - [ ] Add documentation
 - [ ] Updated Python API documentation (swig/include/python/docs/)
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

**Comment by rouault** [comment]
Looks good to me. I just checked that this curl option was added very long time ago in curl 7.49 (https://github.com/curl/curl/commit/324a97ecf82e5e415c3c9fb4df093053c1efedf1), which is fine given our >= 7.68 requirement

**Comment by lnicola** [comment]
Got to say I'm a little confused:

image

Anyway, I smoke-tested this and it seems to work. I don't have an URL to share, though.

File: port/cpl_http.cpp [hunk 1]
```diff
@@ -1150,12 +1150,15 @@ int CPLHTTPPopFetchCallback(void)
  *     and use the first one found as the CAINFO value (GDAL >= 2.1.3). The
  *     GDAL_CURL_CA_BUNDLE environment variable may also be used to set the
  *     CAINFO value in GDAL >= 3.2.</li>
- * <li>HTTP_VERSION=1.0/1.1/2/2TLS (GDAL >= 2.3). Specify HTTP version to use.
+ * <li>HTTP_VERSION=1.0/1.1/2/2TLS (GDAL >= 2.3)/2PRIOR_KNOWLEDGE (GDAL >= TBD).
```
Comment (lnicola) [inline port/cpl_http.cpp:1153]: Would this be 3.10 or 4.0?

And do you have any nits about the phrasing and word wrapping?

**Comment by lnicola** [comment]
Needs a 3.10.0 milestone.

**Pull request merged**